src/hotspot/share/gc/shenandoah/shenandoahHeap.inline.hpp

```diff
@@ -391,7 +391,7 @@ inline bool ShenandoahHeap::is_in_old(const void* p) const {
   return is_in_reserved(p) && (_affiliations[heap_region_index_containing(p)] == ShenandoahAffiliation::OLD_GENERATION);
 }
 
-inline bool ShenandoahHeap::is_old(oop obj) const {
+inline bool ShenandoahHeap::is_not_in_active_young_collection(oop obj) const {
```

Comment:
This sounds to me more like: `is_in_old_when_active_young()`.

The "not" is the name makes the negative of its semantics suspect. Think of the sense of what one might believe if one identified the negation with `is_in_active_young_collection()` which I'd take to be `active_generation()->is_young() && is_in_young(obj)`, rather than the actual negation of the condition which would be: `!active_generation()->is_young() || !is_in_old(obj)`.

Reply:
Can we verify that the predicate is stable in the context of an arbitrary current caller?

Reply:
Active generation only changes on a safepoint. However, generational membership may change during evacuation.